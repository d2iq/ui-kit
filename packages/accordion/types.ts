export type HeadingLevel = 2 | 3 | 4 | 5 | 6;
export type AccordionTitleAppearances = "default" | "danger";
