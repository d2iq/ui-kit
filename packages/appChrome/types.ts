import { BreakpointConfig } from "../shared/styles/breakpoints";
import { SpaceSizes } from "../shared/styles/styleUtils/modifiers/modifierUtils";
import { LinkProps } from "../link/types";
import { IconSize } from "../shared/types/iconSize";

type BgColor = React.CSSProperties["backgroundColor"];
type PaddingHoriz = SpaceSizes;
type PaddingVert = SpaceSizes;

export interface AppChromeTheme {
  headerBackgroundColor?: BgColor;
  headerPaddingHor?: PaddingHoriz;
  headerPaddingVert?: PaddingVert;
  iconWidth?: IconSize;
  itemActiveBackgroundColor?: BgColor;
  itemHoverBackgroundColor?: BgColor;
  sidebarBackgroundColor?: BgColor;
  sidebarHeaderPaddingHor?: PaddingHoriz;
  sidebarHeaderPaddingVert?: PaddingVert;
  sidebarWidth?: BreakpointConfig<React.CSSProperties["width"]>;
  sidebarItemPaddingHor?: PaddingHoriz;
  sidebarItemPaddingVert?: PaddingVert;
}

export interface SidebarNavItemProps extends LinkProps {
  isActive?: boolean;
  onClick?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  disabled?: boolean;
}
