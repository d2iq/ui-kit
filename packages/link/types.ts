import React from "react";

export interface LinkProps {
  /**
   * A url the user will be navigated to when clicking the link. This also changes the tag to `<a>`.
   */
  url?: string;
  /**
   * If the `url` prop is set, this will open that link in a new tab.
   */
  openInNewTab?: boolean;
  children?: React.ReactNode | React.ReactNode[];
}

export type ExpandedLinkProps = LinkProps &
  Omit<React.HTMLProps<HTMLAnchorElement>, "ref">;

export type LinkComponent = React.ComponentType<
  LinkProps & React.HTMLProps<HTMLAnchorElement>
>;
