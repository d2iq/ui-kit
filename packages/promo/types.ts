export interface PromoProps {
  /** The primary banner content */
  bodyContent: React.ReactNode;
  /** A function that gets called when the banner is dismissed */
  dismissHandler?: () => void;
  /** A consise summary of the banner's message */
  headingText: string;
  /** Whether the content is on a dark background. Changes text color to be readable. */
  isDarkBackground?: boolean;
  /** A link to the graphic's image file. The graphic (typically an illustration) should be relevant to the banner's message. Ideally the graphic should not just be decorative, but it reinforces the banner's overall message. */
  graphicSrc?: string;
  /** Whether a user has opted not to see the banner */
  optOutBanner?: boolean;
  /** A function that gets called when the user checks the "Do not show again" checkbox */
  optOutHandler?: (e: React.SyntheticEvent<HTMLInputElement>) => void;
  /** The most relevant action a user can take in response to the banner's message */
  primaryAction?: React.ReactNode;
  /** An additional action a user can take in response to the banner's message. This can also just be a link to more information relevant to the banner's message. */
  secondaryAction?: React.ReactNode;
  /** The Cypress selector */
  ["data-cy"]?: string;
}

export type GradientStyle = "lightBlue" | "purple" | "oceanBlue";
export type GradientColors = [string, string, string?];
export type PromoBackgroundColor = "themeBgSecondary" | "purpleLighten5";
