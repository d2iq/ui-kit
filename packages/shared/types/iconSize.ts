export type IconSize = "xxs" | "xs" | "s" | "m" | "l" | "xl" | "xxl";
