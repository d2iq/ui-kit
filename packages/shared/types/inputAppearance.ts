export enum InputAppearance {
  Standard = "standard",
  Error = "error",
  Success = "success"
}
