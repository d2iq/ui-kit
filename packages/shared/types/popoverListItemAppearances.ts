export enum PopoverListItemAppearances {
  DANGER = "danger"
}
