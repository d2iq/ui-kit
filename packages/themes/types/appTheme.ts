import { AppChromeTheme } from "../../appChrome/types";

type Color = React.CSSProperties["color"];
export type LocalTheme = AppChromeTheme & {
  menuHasIcon: boolean;

  coloredRows: any[];
  mutedRows: any[];

  colors: {
    // Branding
    brandPrimary?: Color;
    brandPrimaryInverted?: Color;
    brandSecondary?: Color;
    brandSecondaryInverted?: Color;

    // States
    success?: Color;
    successInverted?: Color;
    warning?: Color;
    warningInverted?: Color;
    error?: Color;
    errorInverted?: Color;

    // Backgrounds
    bgPrimary?: Color;
    bgPrimaryInverted?: Color;
    bgSecondary?: Color;
    bgSecondaryInverted?: Color;
    bgHover?: Color;
    bgHoverInverted?: Color;
    bgScrim?: Color;
    bgSelected?: Color;
    bgSelectedInverted?: Color;
    bgAppHeader?: Color;

    // Text
    textColorPrimary?: Color;
    textColorPrimaryInverted?: Color;
    textColorSecondary?: Color;
    textColorSecondaryInverted?: Color;
    textColorInteractive?: Color;
    textColorInteractiveInverted?: Color;

    // Decorators
    border?: Color;
    borderInverted?: Color;
    borderHeavy?: Color;
    borderHeavyInverted?: Color;
    shadow?: Color;
    shadowInverted?: Color;
  };
};

declare module "@emotion/react" {
  export interface Theme extends LocalTheme {
    _unused: true; // can not be empty
  }
}
