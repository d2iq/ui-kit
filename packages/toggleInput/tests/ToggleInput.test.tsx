import React from "react";
import { mount } from "enzyme";
import serializer from "@emotion/jest";
import toJson from "enzyme-to-json";
import { ToggleInput } from "../";
import { InputAppearance } from "../../shared/types/inputAppearance";

expect.addSnapshotSerializer(serializer);

describe("ToggleInput", () => {
  it("renders all appearances", () => {
    Object.keys(InputAppearance).forEach(appearance => {
      const component = mount(
        <ToggleInput
          id="defaultId"
          inputLabel="Sample label"
          value="default"
          appearance={InputAppearance[appearance]}
        />
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });
  it("renders a checkbox input type", () => {
    const component = mount(
      <ToggleInput
        id="defaultId"
        inputLabel="Sample label"
        value="default"
        inputType="checkbox"
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders a radio input type", () => {
    const component = mount(
      <ToggleInput
        id="defaultId"
        inputLabel="Sample label"
        value="default"
        inputType="radio"
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("calls onFocus if its passed in as a prop", () => {
    const focusFn = jest.fn();
    const component = mount(
      <ToggleInput
        id="defaultId"
        inputLabel="Sample label"
        value="default"
        onFocus={focusFn}
      />
    );
    expect(focusFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(focusFn).toHaveBeenCalled();
  });

  it("calls onBlur if its passed in as a prop", () => {
    const blurFn = jest.fn();
    const component = mount(
      <ToggleInput
        id="defaultId"
        inputLabel="Sample label"
        value="default"
        onBlur={blurFn}
      />
    );
    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("focus");
    expect(blurFn).not.toHaveBeenCalled();
    component.find("input").simulate("blur");
    expect(blurFn).toHaveBeenCalled();
  });
});
