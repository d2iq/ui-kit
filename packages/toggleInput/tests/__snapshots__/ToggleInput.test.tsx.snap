// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleInput renders a checkbox input type 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-bottom: 0!important;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  display: inherit;
}

.emotion-3 {
  border-color: var(--themeBorder, #DADDE2);
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  border-radius: 4px;
  position: relative;
}

.emotion-4 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-left: 12px;
}

<ForwardRef
  appearance="standard"
  id="defaultId"
  inputLabel="Sample label"
  inputType="checkbox"
  showInputLabel={true}
  value="default"
  vertAlign="center"
>
  <FormFieldWrapper
    id="defaultId"
  >
    <div
      className="emotion-0"
      data-cy="checkboxInput"
    >
      <label
        className="emotion-1"
        htmlFor="defaultId"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-3"
          >
            <input
              aria-describedby=""
              aria-invalid={false}
              className="emotion-4"
              data-cy="checkboxInput-input"
              id="defaultId"
              onBlur={[Function]}
              onFocus={[Function]}
              type="checkbox"
              value="default"
            />
          </div>
        </div>
        <div
          className="emotion-5"
        >
          Sample label
        </div>
      </label>
    </div>
  </FormFieldWrapper>
</ForwardRef>
`;

exports[`ToggleInput renders a radio input type 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-bottom: 0!important;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  display: inherit;
}

.emotion-3 {
  border-color: var(--themeBorder, #DADDE2);
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  border-radius: 999px;
  position: relative;
}

.emotion-4 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-left: 12px;
}

<ForwardRef
  appearance="standard"
  id="defaultId"
  inputLabel="Sample label"
  inputType="radio"
  showInputLabel={true}
  value="default"
  vertAlign="center"
>
  <FormFieldWrapper
    id="defaultId"
  >
    <div
      className="emotion-0"
      data-cy="radioInput"
    >
      <label
        className="emotion-1"
        htmlFor="defaultId"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-3"
          >
            <input
              aria-describedby=""
              aria-invalid={false}
              className="emotion-4"
              data-cy="radioInput-input"
              id="defaultId"
              onBlur={[Function]}
              onFocus={[Function]}
              type="radio"
              value="default"
            />
          </div>
        </div>
        <div
          className="emotion-5"
        >
          Sample label
        </div>
      </label>
    </div>
  </FormFieldWrapper>
</ForwardRef>
`;

exports[`ToggleInput renders all appearances 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-bottom: 0!important;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  display: inherit;
}

.emotion-3 {
  border-color: var(--themeBorder, #DADDE2);
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  border-radius: 4px;
  position: relative;
}

.emotion-4 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-left: 12px;
}

<ForwardRef
  appearance="standard"
  id="defaultId"
  inputLabel="Sample label"
  inputType="checkbox"
  showInputLabel={true}
  value="default"
  vertAlign="center"
>
  <FormFieldWrapper
    id="defaultId"
  >
    <div
      className="emotion-0"
      data-cy="checkboxInput"
    >
      <label
        className="emotion-1"
        htmlFor="defaultId"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-3"
          >
            <input
              aria-describedby=""
              aria-invalid={false}
              className="emotion-4"
              data-cy="checkboxInput-input"
              id="defaultId"
              onBlur={[Function]}
              onFocus={[Function]}
              type="checkbox"
              value="default"
            />
          </div>
        </div>
        <div
          className="emotion-5"
        >
          Sample label
        </div>
      </label>
    </div>
  </FormFieldWrapper>
</ForwardRef>
`;

exports[`ToggleInput renders all appearances 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-bottom: 0!important;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  display: inherit;
}

.emotion-3 {
  border-color: var(--themeBorder, #DADDE2);
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  border-radius: 4px;
  position: relative;
}

.emotion-4 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-left: 12px;
  color: var(--themeError, #EB293A);
  fill: var(--themeError, #EB293A);
}

<ForwardRef
  appearance="error"
  id="defaultId"
  inputLabel="Sample label"
  inputType="checkbox"
  showInputLabel={true}
  value="default"
  vertAlign="center"
>
  <FormFieldWrapper
    id="defaultId"
  >
    <div
      className="emotion-0"
      data-cy="checkboxInput checkboxInput.error"
    >
      <label
        className="emotion-1"
        htmlFor="defaultId"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-3"
          >
            <input
              aria-describedby=""
              aria-invalid={false}
              className="emotion-4"
              data-cy="checkboxInput-input checkboxInput-input.error"
              id="defaultId"
              onBlur={[Function]}
              onFocus={[Function]}
              type="checkbox"
              value="default"
            />
          </div>
        </div>
        <div
          className="emotion-5"
        >
          Sample label
        </div>
      </label>
    </div>
  </FormFieldWrapper>
</ForwardRef>
`;

exports[`ToggleInput renders all appearances 3`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: auto;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-bottom: 0!important;
}

.emotion-1>div {
  width: auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
  display: inherit;
}

.emotion-3 {
  border-color: var(--themeBorder, #DADDE2);
  background-color: var(--themeBgPrimary, #FFFFFF);
  border-style: solid;
  border-width: 1px;
  height: 14px;
  width: 14px;
  border-radius: 4px;
  position: relative;
}

.emotion-4 {
  border: 0;
  clip: rect(0 0 0 0);
  position: absolute;
  overflow: hidden;
  margin: -1px;
  padding: 0;
  width: 1px;
  height: 1px;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
  padding-left: 12px;
  color: var(--themeSuccess, #14C684);
  fill: var(--themeSuccess, #14C684);
}

<ForwardRef
  appearance="success"
  id="defaultId"
  inputLabel="Sample label"
  inputType="checkbox"
  showInputLabel={true}
  value="default"
  vertAlign="center"
>
  <FormFieldWrapper
    id="defaultId"
  >
    <div
      className="emotion-0"
      data-cy="checkboxInput checkboxInput.success"
    >
      <label
        className="emotion-1"
        htmlFor="defaultId"
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-3"
          >
            <input
              aria-describedby=""
              aria-invalid={false}
              className="emotion-4"
              data-cy="checkboxInput-input checkboxInput-input.success"
              id="defaultId"
              onBlur={[Function]}
              onFocus={[Function]}
              type="checkbox"
              value="default"
            />
          </div>
        </div>
        <div
          className="emotion-5"
        >
          Sample label
        </div>
      </label>
    </div>
  </FormFieldWrapper>
</ForwardRef>
`;
